pkg/k8s/watchers/watcher.go

```diff
@@ -516,8 +516,10 @@ func (k *K8sWatcher) EnableK8sWatcher(ctx context.Context) error {
 		go k.ciliumEndpointsInit(ciliumNPClient, asyncControllers)
 	}
 
-	// cilium local redirect policies
-	go k.ciliumLocalRedirectPolicyInit(ciliumNPClient)
+	if option.Config.EnableLocalRedirectPolicy {
```

Comment:
I just noticed, but the other async goroutines for CRD initialization in this function perforrm `asyncControllers.Add(1)` before running a goroutine to initialize the watcher. Presumably this should be done inside the conditional statement here too?

Reply:
Good catch! I didn't mean to run it as a go routine.